geode-core/src/distributedTest/java/org/apache/geode/internal/cache/DeltaForceSizingFlagDUnitTest.java

```diff
@@ -0,0 +1,995 @@
+      // from serialized to deserialized
+      assertTrue(finalPRSize0 - origPRSize0 != 0);
+      assertTrue(finalPRSize1 - origPRSize1 != 0);
+    } else {
+      assertEquals(0, finalPRSize0 - origPRSize0);
+      assertEquals(0, finalPRSize1 - origPRSize1);
+    }
+  }
+
+  private long getSizeFromPRStats(VM vm0) {
+    return (Long) vm0.invoke("getSizeFromPRStats", () -> {
+      Cache cache = ClusterStartupRule.getCache();
+      LocalRegion region = (LocalRegion) cache.getRegion("region");
+      if (region instanceof PartitionedRegion) {
+        long total = 0;
+        PartitionedRegion pr = ((PartitionedRegion) region);
+        for (int i = 0; i < pr.getPartitionAttributes().getTotalNumBuckets(); i++) {
+          total += pr.getDataStore().getBucketSize(i);
+        }
+        return total;
+      } else {
+        return 0L;
+      }
+    });
+  }
+
+  private long getSizeFromEvictionStats(VM vm0) {
+    return (Long) vm0.invoke("getSizeFromEvictionStats", () -> {
+
+      Cache cache = ClusterStartupRule.getCache();
+      LocalRegion region = (LocalRegion) cache.getRegion("region");
+      return getSizeFromEvictionStats(region);
+    });
+  }
+
+  private long getEvictions(VM vm0) {
+    return (Long) vm0.invoke("getEvictions", () -> {
+      Cache cache = ClusterStartupRule.getCache();
+      LocalRegion region = (LocalRegion) cache.getRegion("region");
+      return getEvictions(region);
+    });
+  }
+
+  private int getObjectSizerInvocations(VM vm0) {
+    return (Integer) vm0.invoke("getObjectSizerInvocations", () -> {
+      Cache cache = ClusterStartupRule.getCache();
+      LocalRegion region = (LocalRegion) cache.getRegion("region");
+      return getObjectSizerInvocations(region);
+    });
+  }
+
+  private void assignPRBuckets(MemberVM memberVM) {
+    memberVM.invoke("assignPRBuckets", () -> {
+      Cache cache = ClusterStartupRule.getCache();
+      PartitionRegionHelper.assignBucketsToPartitions(cache.getRegion("region"));
+    });
+  }
+
+  private boolean prHostsBucketForKey(MemberVM memberVM, final Object key) {
+    Boolean result = (Boolean) memberVM.invoke("prHostsBucketForKey", () -> {
+      Cache cache = ClusterStartupRule.getCache();
+      DistributedMember myId = cache.getDistributedSystem().getDistributedMember();
+      Region region = cache.getRegion("region");
+      DistributedMember hostMember = PartitionRegionHelper.getPrimaryMemberForKey(region, key);
+      if (hostMember == null) {
+        throw new IllegalStateException("bucket for key " + key + " is not hosted!");
+      }
+      boolean res = Boolean.valueOf(myId.equals(hostMember));
+      return res;
+    });
+    return result.booleanValue();
+  }
+
+  private void put(VM vm0, final Object key, final Object value) {
+    vm0.invoke("Put data", () -> {
+      Cache cache = ClusterStartupRule.getCache();
+      LocalRegion region = (LocalRegion) cache.getRegion("region");
```

Comment:
Compiler warning can be fixed here by asserting that `cache` is not null before calling methods on it.